sitl_launcher/launch/demo.launch.py

```diff
@@ -0,0 +1,126 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import launch
+from launch import LaunchDescription
+from launch_ros.actions import Node
+from ament_index_python.packages import get_package_share_directory
+from launch.actions import SetEnvironmentVariable
+import sys
+from launch import LaunchService
+from launch_ros import get_default_launch_description
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration
+from launch.conditions import IfCondition
+from launch.event_handlers import OnProcessExit
+from launch import LaunchContext
+from launch.utilities import perform_substitutions, normalize_to_list_of_substitutions
+
+from launch.actions import IncludeLaunchDescription
+from launch import LaunchDescriptionSource
+from launch.substitutions import PythonExpression
+from typing import Dict
+from typing import List
+from typing import Text
+
+class ReplaceWorldString(launch.Substitution):
+  """
+  Substitution that replaces strings on a given file.
+  """
+
+  def __init__(self,
+    sitl_world: launch.SomeSubstitutionsType) -> None:
+    super().__init__()
+
+    from launch.utilities import normalize_to_list_of_substitutions  # import here to avoid loop
+    self.__sitl_world = normalize_to_list_of_substitutions(sitl_world)
+
+  @property
+  def name(self) -> List[launch.Substitution]:
+    """Getter for name."""
+    return self.__sitl_world
+
+  def perform(self, context: launch.LaunchContext) -> Text:
+
+      sitl_world = launch.utilities.perform_substitutions(context, self.name)
+
+      if(sitl_world=="yosemite"):
+          pkg_shared_directory = 'yosemite_valley'
+          launch_file = 'drone_yosemite.launch.py'
+      elif(sitl_world=="mcmillan"):
+          pkg_shared_directory = 'mcmillan_airfield'
+          launch_file = 'drone_mcmillan.launch.py'
+      elif(sitl_world=="ksql"):
+          pkg_shared_directory = 'ksql_airport'
+          launch_file = 'drone_ksql.launch.py'
+      elif(sitl_world=="baylands"):
+          pkg_shared_directory = 'baylands'
+          launch_file = 'drone_baylands.launch.py'
+
+      world_dir = get_package_share_directory(pkg_shared_directory)
+      return world_dir + '/launch/' + launch_file
+
+
+def generate_launch_description():
+
+    declare_sitl_world_cmd = DeclareLaunchArgument(
+        'sitl_world',
+        default_value='yosemite',
+        description='World [yosemite, mcmillan, ksql, baylands]')
+
+    sitl_world = ReplaceWorldString(sitl_world=launch.substitutions.LaunchConfiguration('sitl_world'))
+
+    included_launch = launch.actions.IncludeLaunchDescription(
```

Comment:
I"m not sure were it should be but, the `sitl_world` argument should have a default. Right now it crashes out if unset on the command line.

```
Executing command: 
docker run -it   --rm    -v /home/tfoote:/home/tfoote   --gpus all -v /run/user/1000/pulse:/run/user/1000/pulse --device /dev/snd  -e PULSE_SERVER=unix:/run/user/1000/pulse/native -v /run/user/1000/pulse/native:/run/user/1000/pulse/native --group-add 29   -e DISPLAY -e TERM   -e QT_X11_NO_MITSHM=1   -e XAUTHORITY=/tmp/.docker.xauth -v /tmp/.docker.xauth:/tmp/.docker.xauth   -v /tmp/.X11-unix:/tmp/.X11-unix   -v /etc/localtime:/etc/localtime:ro    038b320bf156 
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-tfoote'
[INFO] [launch]: All log files can be found below /home/tfoote/.ros/log/2019-11-13-13-43-52-395839-c47a23c3448a-10
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): launch configuration 'sitl_world' does not exist
[INFO] [rviz2]: Stereo is NOT SUPPORTED
[INFO] [rviz2]: OpenGl version: 4.6 (GLSL 4.6)
[INFO] [rviz2]: Stereo is NOT SUPPORTED
```

Reply:
This commit solve the issue https://github.com/ahcorde/drone_demo/commit/a11b62ca95aaf6fda5963eb209f3a401ef0388f8